From: Andrew Morton <akpm@linux-foundation.org>
Subject: lib-introduce-hierarchical-per-cpu-counters-fix
Date: Thu Feb 19 08:12:23 PM PST 2026

export things to modules for lib/tests/percpu_counter_tree_kunit.ko

Cc: Mathieu Desnoyers <mathieu.desnoyers@efficios.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 lib/percpu_counter_tree.c |   10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

--- a/lib/percpu_counter_tree.c~lib-introduce-hierarchical-per-cpu-counters-fix
+++ a/lib/percpu_counter_tree.c
@@ -297,6 +297,7 @@ int percpu_counter_tree_init(struct perc
 {
 	return percpu_counter_tree_init_many(counter, items, 1, batch_size, gfp_flags);
 }
+EXPORT_SYMBOL_GPL(percpu_counter_tree_init);
 
 /**
  * percpu_counter_tree_destroy_many() - Destroy many per-CPU counter trees.
@@ -321,6 +322,7 @@ void percpu_counter_tree_destroy(struct
 {
 	return percpu_counter_tree_destroy_many(counter, 1);
 }
+EXPORT_SYMBOL_GPL(percpu_counter_tree_destroy);
 
 static
 long percpu_counter_tree_carry(long orig, long res, long inc, unsigned long bit_mask)
@@ -423,7 +425,7 @@ void percpu_counter_tree_add(struct perc
 		return;
 	percpu_counter_tree_add_slowpath(counter, inc);
 }
-
+EXPORT_SYMBOL_GPL(percpu_counter_tree_add);
 
 static
 long percpu_counter_tree_precise_sum_unbiased(struct percpu_counter_tree *counter)
@@ -435,6 +437,7 @@ long percpu_counter_tree_precise_sum_unb
 		sum += *per_cpu_ptr(counter->level0, cpu);
 	return (long) sum;
 }
+EXPORT_SYMBOL_GPL(percpu_counter_tree_precise_sum_unbiased);
 
 /**
  * percpu_counter_tree_precise_sum() - Return precise counter sum.
@@ -491,6 +494,7 @@ int percpu_counter_tree_approximate_comp
 			     a->approx_accuracy_range.over + b->approx_accuracy_range.under,
 			     a->approx_accuracy_range.under + b->approx_accuracy_range.over);
 }
+EXPORT_SYMBOL_GPL(percpu_counter_tree_approximate_compare);
 
 /**
  * percpu_counter_tree_approximate_compare_value - Approximated comparison of a counter tree against a given value.
@@ -514,6 +518,7 @@ int percpu_counter_tree_approximate_comp
 			     counter->approx_accuracy_range.under,
 			     counter->approx_accuracy_range.over);
 }
+EXPORT_SYMBOL_GPL(percpu_counter_tree_approximate_compare_value);
 
 /**
  * percpu_counter_tree_precise_compare - Precise comparison of two counter trees.
@@ -582,6 +587,7 @@ int percpu_counter_tree_precise_compare(
 		return 1;
 	return 0;
 }
+EXPORT_SYMBOL_GPL(percpu_counter_tree_precise_compare);
 
 /**
  * percpu_counter_tree_precise_compare_value - Precise comparison of a counter tree against a given value.
@@ -619,6 +625,7 @@ int percpu_counter_tree_precise_compare_
 		return 1;
 	return 0;
 }
+EXPORT_SYMBOL_GPL(percpu_counter_tree_precise_compare_value);
 
 static
 void percpu_counter_tree_set_bias(struct percpu_counter_tree *counter, long bias)
@@ -657,6 +664,7 @@ size_t percpu_counter_tree_items_size(vo
 		return 0;
 	return counter_config->nr_items * sizeof(struct percpu_counter_tree_level_item);
 }
+EXPORT_SYMBOL_GPL(percpu_counter_tree_items_size);
 
 static void __init calculate_accuracy_topology(void)
 {
_
